app/src/main/java/zapsolutions/zap/connection/manageWalletConfigs/WalletConfigsManager.java

```diff
@@ -0,0 +1,219 @@
+        }
+    }
+
+    public WalletConfigsJson getWalletConfigsJson() {
+        return mWalletConfigsJson;
+    }
+
+    public static WalletConfigsManager getInstance() {
+        if (mInstance == null) {
+            mInstance = new WalletConfigsManager();
+        }
+
+        return mInstance;
+    }
+
+    private void createEmptyWalletConfigsJson() {
+        mWalletConfigsJson = new Gson().fromJson("{\"connections\":[]}", WalletConfigsJson.class);
+    }
+
+
+    /**
+     * Used to determine if the provided String is a valid walletConfigs JSON.
+     *
+     * @param walletConfigsString parses as JSON
+     * @return if the JSON syntax is valid
+     */
+    public static boolean isValidJson(String walletConfigsString) {
+        try {
+            WalletConfigsJson walletConfigs = new Gson().fromJson(walletConfigsString, WalletConfigsJson.class);
+            return walletConfigs != null;
+        } catch (JsonSyntaxException ex) {
+            return false;
+        }
+    }
+
+
+    /**
+     * Checks if a wallet configuration with the given alias/name already exists.
+     *
+     * @param alias The name of the wallet
+     * @return
+     */
+    public boolean doesWalletConfigExist(@NonNull String alias) {
+
+        return mWalletConfigsJson.getConnection(alias.toLowerCase()) != null;
+    }
+
+
+    /**
+     * Adds a wallet configuration to our current setup.
+     * If a configuration exist with this alias, it will be overridden.
+     * Do not forget to call apply() afterwards to make this change permanent.
+     *
+     * @param alias    Name of the wallet/configuration
+     * @param type     One of the following types: remote, local
+     * @param host     The host
+     * @param port     The port
+     * @param cert     The certificate. This is optional and can be null
+     * @param macaroon The Macaroon. Encoded as base16 (hex)
+     */
+    public void addWalletConfig(@NonNull String alias, String type, String host,
+                                int port, @Nullable String cert, String macaroon) {
+
+        // Create the config
+        WalletConfig config = new WalletConfig();
+        config.setAlias(alias.toLowerCase());
+        config.setType(type);
+        config.setHost(host);
+        config.setPort(port);
+        config.setCert(cert);
+        config.setMacaroon(macaroon);
+
+
+        // Add the config to our configurations array
+        mWalletConfigsJson.addWalletConfig(config);
+
+    }
+
+
+    /**
+     * Returns the wallet config of the currently active wallet.
+     *
+     * @return
+     */
+    public WalletConfig getCurrentWalletConfig() {
+        return getWalletConfig(PrefsUtil.getCurrentWalletConfig());
+    }
+
+
+    /**
+     * Load a wallet configuration by its alias/name.
+     *
+     * @param alias The name of the wallet
+     * @return Returns null if no configuration is found for the given alias
+     */
+    public WalletConfig getWalletConfig(@NonNull String alias) {
+
```

Comment:
I think we can just return `return mWalletConfigsJson.getConnection(alias);`, since it will return null if it does not exist.